scripts/get-resource-index-ref-matrix

```diff
@@ -42,14 +46,19 @@ build_prod_and_canary_ref_matrix() {
     canary_resource_index=$(get_resource_index_at_commit "$canary_commit")
 
     if [[ "$prod_resource_index" == "$canary_resource_index" ]]; then
-        jq_array='[ $PROD_COMMIT ]'
+        jq_array='[ {"ref": $PROD_COMMIT, "resource_index": $PROD_RESOURCE_INDEX} ]'
     else
-        jq_array='[ $PROD_COMMIT, $CANARY_COMMIT ]'
+        jq_array='[
+            {"ref": $PROD_COMMIT, "resource_index": $PROD_RESOURCE_INDEX},
+            {"ref": $CANARY_COMMIT, "resource_index": $CANARY_RESOURCE_INDEX}
+        ]'
     fi
 
     ref_matrix=$(jq -c --null-input \
                     --arg PROD_COMMIT "$prod_commit" \
+                    --arg PROD_RESOURCE_INDEX "$prod_resource_index" \
                     --arg CANARY_COMMIT "$canary_commit" \
+                    --arg CANARY_RESOURCE_INDEX "$canary_resource_index" \
                     "$jq_array")
```

Comment:
_minor, non-blocking_

This could be simplified further a bit by doing the comparison within jq, e.g. something like:

```jq
  [
    {"ref": $PROD_COMMIT, "resource_index": $PROD_RESOURCE_INDEX},
    {"ref": $CANARY_COMMIT, "resource_index": $CANARY_RESOURCE_INDEX}
  ]
| unique_by(.resource_index)
```

Reply:
Also, instead of storing the jq output in `ref_matrix` and then immediately echoing it, the jq command can just be run outside of the process interpolation.

Reply:
Updated, thanks!